### PR TITLE
Add telemetry related to Apps Framework

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Add telemetry related information about the events tracked by the Apps Framework.

#### Related PRs
https://github.com/mattermost/mattermost-plugin-apps/pull/230

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-37324

### CODE REVIEW

File: source/manage/telemetry.rst [hunk 1]
```diff
@@ -195,6 +195,29 @@ The following list details the types of Incident Collaboration metadata we colle
...
+**Data collected for all event types**
+
+- ``PluginVersion``: Version of the plugin.
+- ``ServerVersion``: Version of the server the plugin is running on.
+- ``UserID``: Unique identifier of the server.
+- ``appID``: ID of the App that triggers the event.
+- ``Event``: Type of the event. There are three event types that are tracked: ``install``, ``uninstall``, ``call``, ``oauthComplete``.
+
+**Data collected in ``install`` and ``uninstall`` events**
+- ``appType``: Type of the App installed (http, aws...)
```
Comment (justinegeffen) [inline source/manage/telemetry.rst:212]:
```suggestion
- ``appType``: Type of the App installed (e.g., HTTP, AWS)
```

File: source/manage/telemetry.rst [hunk 2]
```diff
@@ -195,6 +195,29 @@ The following list details the types of Incident Collaboration metadata we colle
...
+
+**Data collected in ``call`` events**
+- ``location``: Call location
+- ``type``: Call type. Right now only submit calls are tracked.
+
+**Data collected in ``oauthComplete`` events**
+- ``UserActualID``: User ID of the user completing the oauth flow.
```
Comment (justinegeffen) [inline source/manage/telemetry.rst:219]:
```suggestion
  - ``UserActualID``: User ID of the user completing the OAuth flow.
```

**Review by justinegeffen — APPROVED** [review]
This is awesome, thank you @larkox!! Just two minor suggestions but I am approving now to expedite the process. :)

**Review by aaronrothschild — APPROVED** [review]
LGTM, good start on tracking events for apps.  

Will this be able to report something such as "Servers, have 2.3 apps installed (average) per server)"?  On the telemetry tracking, are recording everything as events, or is there also "daily summary" elements?

**Comment by larkox** [comment]
@aaronrothschild Right now we are tracking single events. We could track snapshots, but haven't tried so far. Would have to check what approach to give it.

**Comment by aaronrothschild** [comment]
> @aaronrothschild Right now we are tracking single events. We could track snapshots, but haven't tried so far. Would have to check what approach to give it.

I think that is worth some effort to investigate. We should have an ongoing daily snapshot of Apps stats similar to plugins IMO.  Another ticket or this one?

File: source/manage/telemetry.rst [hunk 3]
```diff
@@ -195,6 +195,29 @@ The following list details the types of Incident Collaboration metadata we colle
   
 Error and diagnostic reporting is sent by the client to the endpoint ``api.segment.io``. To opt out, disable the feature in **System Console > Environment > Logging**.
 
+Apps Framework Telemetry
```
Comment (amyblais) [inline source/manage/telemetry.rst:198]: @justinegeffen @larkox There are a few small formatting issues: https://docs.mattermost.com/manage/telemetry.html#apps-framework-telemetry.

(I'm not sure how to fix the backticks not working for bolded letters).

**Pull request merged**